components/squad/squad_row.lua

```diff
@@ -0,0 +1,82 @@
+local Class = require('Module:Class')
+local String = require('Module:String')
+local Player = require('Module:Player')
+
+local _ICON_CAPTAIN = '[[image:Captain Icon.png|18px|baseline|Captain|link=' ..
+	'https://liquipedia.net/rocketleague/Category:Captains|alt=Captain]]'
+local _ICON_SUBSTITUTE = '[[image:Substitution.svg|18px|baseline|Sub|link=|alt=Substitution]]'
+
+local SquadRow = Class.new(
+	function(self, frame, role)
+		self.frame = frame
+		self.content = mw.html.create('tr'):addClass('Player')
+
+		role = string.lower(role or '')
+
+		if role == 'sub' then
+			self.content:addClass('sub')
+		elseif role == 'coach' then
+			self.content:addClass('coach')
+			self.content:css('background-color', '#e5e5e5')
+		elseif role == 'coach/manager' then
+			self.content:addClass('coach/manager')
+			self.content:css('background-color', '#e5e5e5')
+		elseif role == 'coach/substitute' then
+			self.content:addClass('coach/substitute')
```

Comment:
Maybe a `if role:find('coach') then` ?

Reply:
For the colors, maybe. You would end up with very similar code though, so 🤷

Reply:
Ah didn't notice the class'es differed. 
Still could be useful as a general catch all as a fallback (as the last one).

Reply:
we should try to avoid hard coded colors rather move it to css classes

Reply:
Yea this stuff is terrible. Let's leave that for follow-up though.